### ISSUE TITLE
value of type 'any WebSocketClient' has no member 'url'

### ISSUE THREAD

**Issue by nicefella** [mattermost/react-native-network-client#127]
Hi, i received the following error during compiling when i try to run my app. 

I think it may be related to StarScream version. But I am not sure how to fix this by myself.

`
❌  (node_modules/@mattermost/react-native-network-client/ios/WebSocketClient.swift:204:32)

  202 |     func didReceive(event: WebSocketEvent, client: Starscream.WebSocketClient) {
  203 |         
> 204 |         guard let url = client.url() else { return }
      |                                ^ value of type 'any WebSocketClient' has no member 'url'
  205 | 
  206 |         switch event {
  207 |         case .connected(let headers):
`

**Comment by enahum** [comment]
Right.. in fact in our app we are using a specific commit in our Starscream fork see this https://github.com/mattermost/mattermost-mobile/blob/main/ios/Podfile#L60

Now, what would be awesome is to keep the same functionality but go back to upstream Starscream.. not sure if possible though

**Comment by DimanVorosh** [comment]
any updates here?

**Comment by uptoamir** [comment]
any updates here? same issue

**Comment by uptoamir** [comment]
@amyblais this makes building mattermost mobile fail, 
❌  (node_modules/@mattermost/react-native-network-client/ios/WebSocketClient.swift:204:32)

  202 |     func didReceive(event: WebSocketEvent, client: Starscream.WebSocketClient) {
  203 |         
> 204 |         guard let url = client.url() else { return }
      |                                ^ value of type 'any WebSocketClient' has no member 'url'
  205 | 
  206 |         switch event {
  207 |         case .connected(let headers):

❌  (node_modules/@mattermost/react-native-network-client/ios/WebSocketManager.swift:72:95)

  70 |         }
  71 | 
> 72 |         let webSocket = WebSocket(request: request, certPinner: certPinner, clientCredential: clientCredential, compressionHandler: compressionHandler)
     |                                                                                               ^ extra argument 'clientCredential' in call
  73 |         webSocket.delegate = delegate
  74 |         
  75 |         webSockets[url] = webSocket

**Comment by enahum** [comment]
That is weird.

The patches in this folder https://github.com/mattermost/react-native-network-client/tree/master/ios/patches should be applied automatically by installing the pod.

You can check the details on the podspec file
